calico_node/startup.py

```diff
@@ -266,7 +266,11 @@ def main():
         sys.exit(1)
 
     as_num = os.getenv("AS")
-    as_num = as_num or None
+    if os.getenv("USE_HOST_AS") == "true":
```

Comment:
One other detail.   Strictly speaking, if we are using the value configured on the node, then we should really do this by updating the client.create_host() method to not set the AS number in this case.   At the moment there is a potential timing window between reading the current value and then passing it in to create_host() where it will set it again.....

That said, I'd be happy to accept this update as it stands (with the update to use the single environment variable) and I'll raise an issue to close that timing window in a future release.